clang/test/CodeGenSYCL/debug-info-srcpos-kernel.cpp

```diff
@@ -25,7 +25,7 @@ int main() {
 }
 
 // CHECK: define{{.*}} spir_kernel {{.*}}19use_kernel_for_test({{.*}}){{.*}} !dbg [[KERNEL:![0-9]+]] {{.*}}{
-// CHECK: getelementptr inbounds %class.anon, %class.anon addrspace(4)* %{{.*}}, i32 0, i32 0, !dbg [[LINE_A0:![0-9]+]]
+// CHECK: getelementptr inbounds %class.anon, {{.*}}, i32 0, i32 0, !dbg [[LINE_A0:![0-9]+]]
```

Comment:
```suggestion
// CHECK: getelementptr inbounds %class.anon, ptr addrspace(4) %{{.*}}, i32 0, i32 0, !dbg [[LINE_A0:![0-9]+]]
```
That's how I modified other opaque pointer related tests in the past.   This keeps checking the 'addrspace(4)', if that was important to begin with.  Just a suggestion.

Reply:
Thanks for the suggestion.
While we transition to opaque pointers this test is expected to work in both modes. `ptr` check will work only when opaque pointers support is enabled.
Also I don't think that checking second argument type is important for this test. The main goal of this test is to validate debug info metadata.